py2many/inference.py

```diff
@@ -204,8 +243,21 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AST:
         self.generic_visit(node)
 
         node.target.annotation = node.annotation
-        if get_id(node.annotation) in self.FIXED_WIDTH_INTS_NAME:
+        target = node.target
+        target_typename = self._clike._typename_from_annotation(target)
+        if target_typename in self.FIXED_WIDTH_INTS_NAME:
             self.has_fixed_width_ints = True
+        annotation = get_inferred_type(node.value)
+        value_typename = self._clike._generic_typename_from_type_node(annotation)
+        target_class = class_for_typename(target_typename, None)
+        value_class = class_for_typename(value_typename, None)
+        if (
+            not is_compatible(target_class, value_class, target, node.value)
+            and target_class != None
+        ):
+            raise Exception(
```

Comment:
It would be nice if this raised `NotImplementedError` and had a snippet in `test_unsupported` which triggers this.

Reply:
Agreed. Also `node.lineno` is good information to add to all the `NotImplementedError` in the code base so the author of python can quickly track down where the error is coming from.

Reply:
Actually `TypeError` makes more sense here, since there is nothing we can do to implement this.